odoo/tools/mail.py

```diff
@@ -283,6 +297,31 @@ def is_html_empty(html_content):
     tag_re = re.compile(r'\<\s*\/?(?:p|div|span|br|b|i)(?:(?=\s+\w*)[^/>]*|\s*)/?\s*\>')
     return not bool(re.sub(tag_re, '', html_content).strip())
 
+def parse_small_qweb(text):
+    reg = re.compile(r"""
+    (
+        (?:
+            (?:\\\\)
+            |(?:\\\$)
+            |(?:(?!\$\{).)
+        )*
+    )
+    (?:\$\{
+        ((?:[^}\'\"\\]
+            |'(?:[^'\\]|\\.)*'?
+            |"(?:[^"\\]|\\.)*"?
+            |(?:\\.))*)
+    })?
+    """, re.X | re.DOTALL)
+
+    groups = []
+    for group in reg.findall(text)[0:-1]:
+        string_match = re.sub(r'\\\\', r'\\', group[0])
+        expression_match = re.sub(r'(?<!\\)\\}', r'}', group[1])
+        expression_match = re.sub(r'\\\\', r'\\', expression_match)
+        groups.append((string_match, expression_match))
+    return groups
```

Comment:
Seems like it would be easier to use `iterfind` and remember the offsets of the previous group in order to extract the intermediate content?

Reply:
It is not an xml file but a text file. Did I miss something ?

Reply:
> It is not xml data but text data. Did I miss something ?

@xmo-odoo was probably talking about [`re.finditer()`](https://docs.python.org/3/library/re.html#re.finditer)